etc/php/8.1/fpm/pool.d/01-deskpro.conf.tmpl

```diff
@@ -9,7 +9,7 @@ listen.mode = 0666
 pm.status_listen = /run/php_fpm_{{.}}.status.sock
 pm.status_path = /fpm/status
 
-clear_env = yes
+clear_env = {{ getenv "DISABLE_CLEAR_ENV" "false" | ternary "no" "yes" }}
```

Comment:
Do we know what specifically actually prevents us from being able to clear the env?

There are a few ways to get around it:

* Write the env "something" needs to an actual config file that gets set at boot time
* If it's something privileges that runs, then you can [re-hydrate the env like done here](https://github.com/deskpro/docker-product-base/blob/main/usr/local/bin/print-container-vars#L24)

Note that the only env that gets cleared is specifically the vars set in the [var reference file](https://github.com/deskpro/docker-product-base/blob/main/usr/local/share/deskpro/container-var-reference.json) so it shouldn't affect other stuff.

Reply:
Sortof similar but I guess solving this would also mean the built-in utils like [mysql-primary](https://github.com/deskpro/docker-product-base/blob/main/usr/local/bin/mysql-primary) would 'just work' the same

Reply:
NVM I misread this, this is just about clearing env in php land.